src/main/kotlin/com/github/gradle/node/task/NodeTask.kt

```diff
@@ -3,54 +3,56 @@ package com.github.gradle.node.task
 import com.github.gradle.node.NodePlugin
 import com.github.gradle.node.exec.NodeExecConfiguration
 import com.github.gradle.node.exec.NodeExecRunner
-import groovy.lang.Closure
+import org.gradle.api.Action
 import org.gradle.api.DefaultTask
 import org.gradle.api.tasks.*
 import org.gradle.api.tasks.PathSensitivity.RELATIVE
-import org.gradle.kotlin.dsl.invoke
+import org.gradle.kotlin.dsl.listProperty
+import org.gradle.kotlin.dsl.mapProperty
+import org.gradle.kotlin.dsl.property
 import org.gradle.process.ExecSpec
-import java.io.File
 
 open class NodeTask : DefaultTask() {
     @get:InputFile
     @get:PathSensitive(RELATIVE)
-    var script: File? = null
+    val script = project.objects.fileProperty()
 
     @get:Input
-    var options = listOf<String>()
+    val options = project.objects.listProperty<String>()
 
     @get:Input
-    var args = listOf<String>()
+    val args = project.objects.listProperty<String>()
 
     @get:Input
-    var ignoreExitValue = false
+    val ignoreExitValue = project.objects.property<Boolean>().convention(false)
 
     @get:Internal
-    var workingDir: File? = null
+    val workingDir = project.objects.directoryProperty()
 
     @get:Input
-    var environment: Map<String, String> = mapOf()
+    val environment = project.objects.mapProperty<String, String>()
 
     @get:Internal
-    var execOverrides: (ExecSpec.() -> Unit)? = null
+    val execOverrides = project.objects.property<Action<ExecSpec>>()
 
     init {
         group = NodePlugin.NODE_GROUP
         dependsOn(NodeSetupTask.NAME)
     }
 
-    // For Groovy DSL
+    // For DSL
     @Suppress("unused")
-    fun setExecOverrides(execOverrides: Closure<ExecSpec>) {
-        this.execOverrides = { execOverrides.invoke(this) }
+    fun execOverrides(execOverrides: Action<ExecSpec>) {
+        this.execOverrides.set(execOverrides)
     }
 
     @TaskAction
     fun exec() {
-        val currentScript = checkNotNull(script) { "Required script property is not set." }
-        val command = options.plus(currentScript.absolutePath).plus(args)
+        val currentScript = script.get().asFile
```

Comment:
This can't be null nowadays, right?

Reply:
`Property.get()` returns a `IllegalStateException` when the value it not set. It was required before and is still required now. We just no longer have a custom error message.

I don't know what are Gradle recommendations regarding this question.